src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -377,6 +258,220 @@ private void basicAck(Channel channel, long deliveryTag) throws IOException {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleReturnListenerException(Channel channel, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleConfirmListenerException(Channel channel, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleBlockedListenerException(Connection connection, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleConsumerException(Channel channel, Throwable exception, Consumer consumer, String consumerTag, String methodName) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleConnectionRecoveryException(Connection conn, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleChannelRecoveryException(Channel ch, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleTopologyRecoveryException(Connection conn, Channel ch, TopologyRecoveryException exception) {
+                turnOffReadiness(exception);
+            }
+
+            private void turnOffReadiness(Throwable exception){
+                switch (connectionName) {
+                    case PUBLISH_CONNECTION_NAME:
+                        publishMetrics.getReadinessMonitor().disable();
+                        break;
+                    case CONSUME_CONNECTION_NAME:
+                        consumeMetrics.getReadinessMonitor().disable();
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + connectionName);
+                }
+                LOGGER.debug("Set RabbitMQ readiness to false. RabbitMQ error", exception);
+            }
+        });
+
+        factory.setConnectionRecoveryTriggeringCondition(shutdownSignal -> !connectionIsClosed.get() && getConnectionRecoveryTriggeringCondition(connectionName, onFailedRecoveryConnection));
+
+        factory.setRecoveryDelayHandler(recoveryAttempts -> getRecoveryDelay(connectionName));
+
+
+
+        HealthMetrics tmpMetrics;
+        switch (connectionName) {
+            case PUBLISH_CONNECTION_NAME:
+                tmpMetrics = publishMetrics;
+                factory.setSharedExecutor(publishSharedExecutor);
+                break;
+            case CONSUME_CONNECTION_NAME:
+                tmpMetrics = consumeMetrics;
+                factory.setSharedExecutor(consumeSharedExecutor);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + connectionName);
+        }
```

Comment:
Extract lines 324-336 to method `getHealthMetrics()` like `getRecoveryListener()`?

Reply:
not actual